other_procore_apis/bim_web_viewer.md

```diff
@@ -839,6 +796,540 @@ undefined
+| floorPlanPoint            | function    | Provides a callback function to calculate a point on the current floor plan. Returns a promise.             |
+| overlay                   | HTMLElement | The HTMLElement for the 2D Navigation overlay                                                               |
+| type                      | string      | The type of interaction and has the following values: `open`, `close`, and `update`                         |
+
+### floorPlanNavigateAndClose
+
+<p class="heading-link-container"><a class="heading-link" href="#floorplannavigateandclose"></a></p>
+
+Updates the camera position and direction in addition to dismissing the 2D floor plan overlay.
+
+#### Parameters
+
+| Field Name | Required | Type   | Description                |
+| ---------- | -------- | ------ | -------------------------- |
+| x          | true     | number | x coordinate in 3D space.  |
+| y          | true     | number | y coordinate in 3D space.  |
+| z          | true     | number | z coordinate in 3D space.  |
+| dx         | true     | number | x direction of the camera. |
+| dy         | true     | number | y direction of the camera. |
+
+### floorPlanPoint
+
+<p class="heading-link-container"><a class="heading-link" href="#floorplanpoint"></a></p>
+
+Returns the corresponding point on the 2D floor plan overlay from a point in 3D model space.
+
+#### Parameters
+
+| Field Name | Required | Type   | Description                 |
+| ---------- | -------- | ------ | --------------------------- |
+| x          | true     | number | x coordinate of the camera. |
+| y          | true     | number | y coordinate of the camera. |
+| z          | true     | number | z coordinate of the camera. |
+| dx         | true     | number | x direction of the camera.  |
+| dy         | true     | number | y direction of the camera.  |
+
+#### Example
+
+```js
+const viewer = new ProcoreBim.Webviewer(options);
+
+let floorPlanNavigateAndClose = null;
+
+viewer.events.addEventListener('floorPlan', (data) => {
+  // Executed when the 2D floor plan overlay is interacted with.
+  if (data.type === 'open') {
+    // Your application might need to know when the 2D floor plan overlay is `open` and keep a reference to
+    // floorPlanNavigateAndClose
+    floorPlanNavigateAndClose = data.floorPlanNavigateAndClose;
+  }
+});
+viewer.start();
+
+// Perhaps your application places markers that represent 3D positions on the 2D floor plan overlay while it is open. When
+// tapping on a marker, this function could be called that uses `floorPlanNavigateAndClose` to set the camera and close the
+// 2D floor plan overlay.
+const goto = (camera) => {
+  if (floorPlanNavigateAndClose !== null) {
+    floorPlanNavigateAndClose(camera.x, camera.y, camera.z, 0.1, 0.1);
+    floorPlanNavigateAndClose = null;
+  }
+};
+```
```

Comment:
This is maybe starting to get nitpicky, but I found it confusing that `floorPlanNavigateAndClose` is at the same level of hierarchy as `floorPlan`, which makes it look like an actual event as opposed to a parameter of the `floorPlan` callback. Maybe we could make `floorPlanNavigateAndClose` and `floorPlanPoint` (and the headers under them) one level deeper.

I'll leave individual line suggestions to make it clearer what I'm proposing to change.

Reply:
Yes! This is great. I was wondering about this too, I think this is a perfect solution to this.